src/app/src/main/kotlin/org/imperial/mrc/hint/exceptions/HintExceptionHandler.kt

```diff
@@ -111,12 +114,74 @@ class HintExceptionHandler(private val errorCodeGenerator: ErrorCodeGenerator,
                 .toResponseEntity()
     }
 
-}
+    private fun viewErrorPage(page: String): ModelAndView
+    {
+        return ModelAndView(page)
+    }
+
+    // resource bundles are encoded with ISO-8859-1
+    fun ResourceBundle.getUTF8String(key: String): String
+    {
+        return this.getString(key)
+                .toByteArray(Charsets.ISO_8859_1)
+                .toString(Charsets.UTF_8)
+    }
+
+    fun handleOtherExceptions(error: Throwable?, request: WebRequest): ResponseEntity<Any>
```

Comment:
Very nearly there, looking good! But I think we should pull out a method here `handleSpringException` which deals with all the specific errors below, and includes the original error message in the `unexpectedError`:

```
        // this handles standard Spring MVC exceptions which do not contain	
        // sensitive info so we return the original error message here	
        return unexpectedError(status, request, e.message)
```

whereas in the else clause, for all other exceptions, we don't include the original message.